daemon/daemon_main.go

```diff
@@ -1421,32 +1431,56 @@ func initKubeProxyFreeOptions() {
 		option.Config.EnableHostServicesUDP = true
 	}
 
-	if option.Config.EnableNodePort && option.Config.Device == "undefined" {
-		device, err := linuxdatapath.NodeDeviceNameWithDefaultRoute()
-		if err != nil {
-			log.WithError(err).Fatal("BPF NodePort's external facing device could not be determined. Use --device to specify.")
-		}
-		log.WithField(logfields.Interface, device).
-			Info("Using auto-derived device for BPF node port")
-		option.Config.Device = device
-	}
-
 	if option.Config.EnableHostReachableServices {
 		if option.Config.EnableHostServicesTCP &&
 			(option.Config.EnableIPv4 && bpf.TestDummyProg(bpf.ProgTypeCgroupSockAddr, bpf.BPF_CGROUP_INET4_CONNECT) != nil ||
 				option.Config.EnableIPv6 && bpf.TestDummyProg(bpf.ProgTypeCgroupSockAddr, bpf.BPF_CGROUP_INET6_CONNECT) != nil) {
-			log.Fatal("BPF host reachable services for TCP needs kernel 4.17.0 or newer.")
+			if strict {
+				log.Fatal("BPF host reachable services for TCP needs kernel 4.17.0 or newer.")
+			} else {
+				log.Warn("BPF host reachable services for TCP needs kernel 4.17.0 or newer. Disabling the feature.")
+				option.Config.EnableHostServicesTCP = false
+			}
 		}
-		// NOTE: as host-lb is a hard dependency for NodePort BPF, the following
-		//       probe will catch if the fib_lookup helper is missing (< 4.18),
-		//       which is another hard dependency for NodePort BPF.
 		if option.Config.EnableHostServicesUDP &&
 			(option.Config.EnableIPv4 && bpf.TestDummyProg(bpf.ProgTypeCgroupSockAddr, bpf.BPF_CGROUP_UDP4_RECVMSG) != nil ||
 				option.Config.EnableIPv6 && bpf.TestDummyProg(bpf.ProgTypeCgroupSockAddr, bpf.BPF_CGROUP_UDP6_RECVMSG) != nil) {
-			log.Fatal("BPF host reachable services for UDP needs kernel 4.19.57, 5.1.16, 5.2.0 or newer. If you run an older kernel and only need TCP, then specify: --host-reachable-services-protos=tcp")
+			if strict {
+				log.Fatal("BPF host reachable services for UDP needs kernel 4.19.57, 5.1.16, 5.2.0 or newer. If you run an older kernel and only need TCP, then specify: --host-reachable-services-protos=tcp")
+			} else {
+				log.Warn("BPF host reachable services for UDP needs kernel 4.19.57, 5.1.16, 5.2.0 or newer. Disabling the feature.")
+				option.Config.EnableHostServicesUDP = false
+			}
```

Comment:
If both are false we should also set `option.Config.EnableHostReachableServices` to false.